lib/glimesh/community_team.ex

```diff
@@ -0,0 +1,124 @@
+defmodule Glimesh.CommunityTeam do
+  @moduledoc """
+  The Community Team context :)
+  """
+  import Ecto.Query, warn: false
+  alias Glimesh.CommunityTeam.AuditLog
+  alias Glimesh.Repo
+
+  def access_level_to_title(level) do
+    case get_global_access_level(level) do
+      5 -> "Admin"
+      4 -> "Manager"
+      3 -> "Team Lead"
+      2 -> "Team Member"
+      1 -> "Trial Member"
+      _ -> "None"
+    end
+  end
+
+  def can_edit_user(user) do
+    if get_global_access_level(user.gct_level) >= 3, do: true, else: false
+  end
+
+  def can_edit_channel(user) do
+    if get_global_access_level(user.gct_level) >= 2, do: true, else: false
+  end
+
+  def can_edit_user_profile(user) do
+    if get_global_access_level(user.gct_level) >= 2, do: true, else: false
+  end
+
+  def can_ban_user(user) do
+    if get_global_access_level(user.gct_level) >= 2, do: true, else: false
+  end
+
+  def can_view_audit_log(user) do
+    if get_global_access_level(user.gct_level) >= 3, do: true, else: false
+  end
+
+  def can_view_billing_info(user) do
+    if get_global_access_level(user.gct_level) >= 4, do: true, else: false
+  end
+
+  def create_audit_entry(user, attrs \\ %{action: "None", target: "None"}) do
+    %AuditLog{
+      user: user
+    }
+    |> AuditLog.changeset(attrs)
+    |> Repo.insert()
+  end
+
+  def list_all_audit_entries(include_verbose? \\ false, params \\ []) do
+    _entries =
+      case include_verbose? do
+        true ->
+          AuditLog |> order_by(desc: :inserted_at) |> preload(:user) |> Repo.paginate(params)
+
+        false ->
+          AuditLog
+          |> order_by(desc: :inserted_at)
+          |> where([al], al.verbose_required? == false)
+          |> preload(:user)
+          |> Repo.paginate(params)
+      end
+  end
+
+  def get_audit_log_entry_from_id!(id) do
+    Repo.one(
+      from al in AuditLog,
+      where: al.id == ^id
+    ) |> Repo.preload([:user])
+  end
+
+  def get_global_access_level(level_int) do
+    if level_int, do: Enum.at(Integer.digits(level_int), 0, 0), else: 0
+  end
+
+  def get_billing_override(level_int) do
+    if level_int, do: Enum.at(Integer.digits(level_int), 1, 0), else: 0
+  end
+
+  def generate_update_user_profile_more_details(user, user_params) do
```

Comment:
For these(since I know they are terribly ugly), the plan is to eventually generate these strings dynamically based on the changed data. Most likely will need to compare the two maps and then find the differences. This was just a quick way of getting the needed information without spending too long on it.